app_dart/lib/src/service/build_status_provider.dart

```diff
@@ -122,6 +124,31 @@ class BuildStatusService {
     }
   }
 
+  /// This function gets called with the last 40 builds fo a given builder ordered
+  /// by creation time starting with the last one first. [tasks] has a list of tasks
+  /// retrieved from BuildBucket using RPCs and [log] is the AppEngine logger to be
+  /// able to logging messages.
+  Future<String> latestLUCIStatus(List<LuciTask> tasks, Logging log) async {
+    for (LuciTask task in tasks) {
+      if (task.ref != 'refs/heads/master') {
+        log.debug('Skipping ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+        continue;
+      }
+      switch (task.status) {
+        case Task.statusFailed:
+        case Task.statusInfraFailure:
+          log.debug('Using ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+          return GithubBuildStatusUpdate.statusFailure;
+        case Task.statusSucceeded:
+          log.debug('Using ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+          return GithubBuildStatusUpdate.statusSuccess;
+      }
+    }
+    // No state means we don't have a state for the last 40 commits which should
+    // close the tree.
+    return GithubBuildStatusUpdate.statusFailure;
```

Comment:
I'm concerned defaulting to this will make it harder to rename builders

Reply:
Can you explain why?

Reply:
Please note that this will be taking into account only active builders as returned by scheduler.

Reply:
Looking at it again, it shouldn't be a concern. For example, in flutter/flutterI need to rename `framework_tests_misc` to `Windows framework_tests_misc` to make the name match the builder name. However, this is aggregating by the builder name for now.